### ISSUE TITLE
bug: useContractReads doesn't work on polymorphic functions

### ISSUE THREAD

**Issue by saurabhburade** [wagmi-dev/wagmi#2636]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.2.1

### Current Behavior

Gives error
```
ContractFunctionExecutionError: The contract function "addr" returned no data ("0x").

This could be due to any of the following:
  - The contract does not have the function "addr",
  - The parameters passed to the contract function may be invalid, or
  - The address is not a contract.
 
Contract Call:
  address:   0x4976fb03C32e5B8cfe2b6cCB31c09Ba78EBaBa41
  function:  addr(bytes32 node)
  args:          (0x4e34d3a81dc3a20f71bbdf2160492ddaa17ee7e5523757d47153379c13cb46df)

Docs: https://viem.sh/docs/contract/multicall.html
Version: viem@0.3.50
    at getContractError (webpack-internal:///./node_modules/viem/dist/esm/utils/errors/getContractError.js:31:12)
    at eval (webpack-internal:///./node_modules/viem/dist/esm/actions/public/multicall.js:109:110)
    at Array.map (<anonymous>)
    at multicall (webpack-internal:///./node_modules/viem/dist/esm/actions/public/multicall.js:92:27)
    at async Object.queryFn (webpack-internal:///./src/hooks/usePublicResolver.ts:146:20)
```

### Expected Behavior

It should
- Resolve the args with respect to function signatures 
- Should follow polymorphic functions

Case 1 
https://etherscan.io/address/0x4976fb03C32e5B8cfe2b6cCB31c09Ba78EBaBa41#readContract
Here I want to check the address records for certain coin-type, see below as the expected contract call

```
Contract Call:
  address:   0x4976fb03C32e5B8cfe2b6cCB31c09Ba78EBaBa41
  function:  addr(bytes32 node,uint256 coinType)
  args:          (0x4e34d3a81dc3a20f71bbdf2160492ddaa17ee7e5523757d47153379c13cb46df,60)
  
Expected Op :   0xfe89cc7abb2c4183683ab71653c4cdc9b02d44b7
  
```

Case 2
https://etherscan.io/address/0x4976fb03C32e5B8cfe2b6cCB31c09Ba78EBaBa41#readContract
Here I want to check the addr only, see below as the expected contract call

```
Contract Call:
  address:   0x4976fb03C32e5B8cfe2b6cCB31c09Ba78EBaBa41
  function:  addr(bytes32 node)
  args:          (0x4e34d3a81dc3a20f71bbdf2160492ddaa17ee7e5523757d47153379c13cb46df)
  
Expected Op : 0xFe89cc7aBB2C4183683ab71653C4cdc9B02D44b7
  
```

### Steps To Reproduce

```
useContractReads({
    chainId: 1,
    contracts: [
       {
          address: '0x4976fb03C32e5B8cfe2b6cCB31c09Ba78EBaBa41',
          abi: PublicResolver, // Add ABI here
          functionName: "addr",
          args: ["0x4e34d3a81dc3a20f71bbdf2160492ddaa17ee7e5523757d47153379c13cb46df", "60"],
       }
      ],
      multicallAddress: "0xcA11bde05977b3631167028862bE2a173976CA11",
  });
```

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://stackblitz.com/edit/new-wagmi-qsudbm?file=src%2FApp.tsx,src%2Fmain.tsx

**Comment by jxom** [comment]
The cointype is of type `bigint`. You are passing in a `string`.

**Comment by saurabhburade** [comment]
It still doesn't work even if I pass the typed param, it gives an unexpected response.
Reproduced here:  https://stackblitz.com/edit/new-wagmi-qsudbm

Expected : ` {result: '0xfe89cc7abb2c4183683ab71653c4cdc9b02d44b7', status: 'success'}
`
Current : `{result: '0x0000000000000000000000000000000000000020', status: 'success'}`

**Comment by jxom** [comment]
That's weird! Taking a look.

**Comment by jxom** [comment]
Should be fixed in viem@1.2.1
